ta/pkcs11/src/persistent_token.c

```diff
@@ -592,8 +594,11 @@ struct ck_token *init_persistent_db(unsigned int token_id)
 
 	db_main = TEE_Malloc(sizeof(*db_main), TEE_MALLOC_FILL_ZERO);
 	db_objs = TEE_Malloc(sizeof(*db_objs), TEE_MALLOC_FILL_ZERO);
-	if (!db_main || !db_objs)
+	if (!db_main || !db_objs) {
+		res = PKCS11_CKR_DEVICE_MEMORY;
```

Comment:
Actually this change is not needed. `res` content is discarded when the function returns. Maybe an alternative is the print an error message here. Ditto for the changes at lines 625 and 645.

Note that this change is unrelated to the change described in the commit message. If you want to add such explicit error messages, could you move the change for these 3 lines (598, 625 and 645) to a specific commit like `ta: pkcs11: improve error trace in token initialization`?

Reply:
okay, it should be fine if the `res` will not be checked after function returns. Let me exclude this trace improvement.